client/src/main/java/pro/panopticon/client/sensor/impl/SuccessrateSensor.java

```diff
@@ -60,57 +60,82 @@ public synchronized void tickFailure(String key) {
-                .map((Map.Entry<AlertInfo, CircularFifoQueue<Event>> e) -> {
+                .map((Map.Entry<AlertInfo, CircularFifoQueue<Tick>> e) -> {
                     AlertInfo alertInfo = e.getKey();
-                    List<Event> events = e.getValue().stream().collect(toList());
+                    List<Tick> events = new ArrayList<>(e.getValue());
                     int all = events.size();
-                    long success = events.stream().filter(a -> a == Event.SUCCESS).count();
-                    long failure = events.stream().filter(a -> a == Event.FAILURE).count();
+                    long success = events.stream().filter(tick -> tick.event == Event.SUCCESS).count();
+                    long failure = events.stream().filter(tick -> tick.event == Event.FAILURE).count();
                     double percentFailureDouble = all > 0 ? (double) failure / (double) all : 0;
                     boolean enoughDataToAlert = all == numberToKeep;
-                    String display = String.format("Last %s calls: %s success, %s failure (%.2f%% failure)%s",
+                    boolean allTicksAreTooOld = allTicksAreTooOld(events);
+                    String display = String.format("Last %s calls: %s success, %s failure (%.2f%% failure)%s%s",
                             Integer.min(all, numberToKeep),
                             success,
                             all - success,
                             percentFailureDouble * 100,
-                            enoughDataToAlert ? "" : " - not enough calls to report status yet"
+                            enoughDataToAlert ? "" : " - not enough calls to report status yet",
+                            allTicksAreTooOld ? "" : " - all ticks are outdated"
                     );
-                    return new Measurement(alertInfo.getSensorKey(), getStatusFromPercentage(enoughDataToAlert, percentFailureDouble), display, new Measurement.CloudwatchValue(percentFailureDouble * 100, StandardUnit.Percent), alertInfo.getDescription());
+                    String status = decideStatus(enoughDataToAlert, percentFailureDouble, allTicksAreTooOld);
+                    return new Measurement(alertInfo.getSensorKey(), status, display, new Measurement.CloudwatchValue(percentFailureDouble * 100, StandardUnit.Percent), alertInfo.getDescription());
                 })
                 .collect(toList());
     }
 
-    private String getStatusFromPercentage(boolean enoughDataToAlert, double percentFailure) {
+    private String decideStatus(boolean enoughDataToAlert, double percentFailure, boolean allTicksAreTooOld) {
         if (!enoughDataToAlert) return "INFO";
+        if (allTicksAreTooOld) return "INFO";
         if (errorLimit != null && percentFailure >= errorLimit) return "ERROR";
         if (warnLimit != null && percentFailure >= warnLimit) return "WARN";
         return "INFO";
     }
 
+    private boolean allTicksAreTooOld(List<Tick> events) {
```

Comment:
Selv om det ikke er grammatisk korrekt ville jeg ha vurdert å kalle denne metoden (og tilhørende variabler) `isAllTicksTooOld`. Det er ganske vanlig i Java å begynne sånne boolean-metoder og variabler med `is*`. Blir veldig lett å se hva slags verdi det er. Men det er ikke strengt tatt nødvendig altså 😄